src/main/java/software/amazon/smithy/lsp/SmithyTextDocumentService.java

```diff
@@ -332,8 +345,11 @@ private List<String> textBufferContents(String path) throws IOException {
         return contents;
     }
 
-    private String findToken(String path, Position p) throws IOException {
+    private Optional<String> findToken(String path, Position p) throws IOException {
```

Comment:
Can this fix for the `findToken` on empty documents be moved to its own PR? This PR is trying to do a lot with this bundled in with the CodeAction changes.

Reply:
yeah, do you mind if rebase to remove the relevant commits?

Reply:
That works.